libopenage/presenter/presenter.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019-2023 the openage authors. See copying.md for legal info.
+
+#include "presenter.h"
+
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "engine/engine.h"
+#include "log/log.h"
+#include "renderer/camera/camera.h"
+#include "renderer/entity_factory.h"
+#include "renderer/gui/gui.h"
+#include "renderer/gui/integration/public/gui_application_with_logger.h"
+#include "renderer/gui/qml_info.h"
+#include "renderer/resources/shader_source.h"
+#include "renderer/resources/texture_info.h"
+#include "renderer/stages/screen/screen_renderer.h"
+#include "renderer/stages/skybox/skybox_renderer.h"
+#include "renderer/stages/terrain/terrain_renderer.h"
+#include "renderer/stages/world/world_renderer.h"
+#include "renderer/window.h"
+#include "util/path.h"
+
+namespace openage::presenter {
+
+Presenter::Presenter(const util::Path &root_dir,
+                     const std::shared_ptr<engine::Engine> engine) :
```

Comment:
missing `&`, although it doesn't matter here.

Reply:
fixed